app/src/main/java/org/tuerantuer/launcher/ui/components/ScrollableLazyGrid.kt

```diff
@@ -0,0 +1,62 @@
+package org.tuerantuer.launcher.ui.components
+
+import androidx.compose.foundation.gestures.detectVerticalDragGestures
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyGridScope
+import androidx.compose.foundation.lazy.grid.LazyGridState
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.nestedscroll.NestedScrollConnection
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.input.pointer.consumeAllChanges
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun ScrollableLazyVerticalGrid(
+    columns: GridCells,
+    modifier: Modifier = Modifier,
+    scrollEnabled: Boolean,
+    state: LazyGridState,
+    contentPadding: PaddingValues = PaddingValues(0.dp),
+    content: LazyGridScope.() -> Unit,
+) {
+    val appliedModifier = modifier.then(
+        if (scrollEnabled) {
+            Modifier // Normal modifier when scrolling is enabled
+        } else {
+            Modifier
+                .disableScrolling()
+                .nestedScroll(NoOpNestedScrollConnection) // Apply no-op nested scroll connection
+        }
+    )
```

Comment:
```suggestion
val appliedModifier = if (scrollEnabled) {
        modifier // Normal modifier when scrolling is enabled
    } else {
        modifier
            .disableScrolling()
            .nestedScroll(NoOpNestedScrollConnection) // Apply no-op nested scroll connection
    }
```
This should do the same thing but simpler